src/app/qgsapplayertreeviewmenuprovider.cpp

```diff
@@ -254,11 +250,24 @@ QMenu *QgsAppLayerTreeViewMenuProvider::createContextMenu()
         menu->addAction( actions->actionShowFeatureCount( menu ) );
 
         menu->addSeparator();
+
+        // save as vector file
+        QMenu *menuExportVector = new QMenu( tr( "Export" ), menu );
+        menuExportVector->addAction( tr( "Save as…" ), QgisApp::instance(), SLOT( saveAsFile() ) );
```

Comment:
Can you please make the actions first, and then setup the connection? Then we can avoid the fragile old-style connects used here.